Update csproj to latest format and target netstandard1.0 and PCL (portable-net45+win8+wp8)

Mark-RSK opened the pull request:
I have updated all nuget references and included the latest Xamarin.Forms that has a netstandard target. To allow the project to build correctly for netstandard and use the inbuilt nuget packing in VS2017 I have moved the IOC files into the main project.

I have tested bringing in the compiled nuget into both a netstandard and PCL XF core project and both function correctly.

Your thoughts on the implementation would be much appreciated as I have attempted this as a way of learning the new csproj structure

Mark-RSK commented:
#160

Mark-RSK commented:
For a working netstandard Xamarin Forms template the following link has a great explanation, and also a repo with the template included.

https://oren.codes/2017/04/23/using-xamarin-forms-with-net-standard-vs-2017-edition/

WiteCastle commented:
I think the line count reduction of ~11,000 lines speaks to Oren's idea that .NET Standard reduces "gunk".

JimSEOW commented:
@Mark-RSK  I came here because I need FreshMVVM to work with Rx.NET with .NET standards 1.6 support.
The project is changed to 1.4 to make it compatible with Xamarin Forms UWP.

JimSEOW commented:
[[FreshMVVm + ReactiveUI](https://github.com/stather/FreshMvvmRxUiInterop) ]+  ReactiveExtension

Mark-RSK commented:
@JimSEOW I have updated the project to use the PackageTargetFallback attribute in the csproj so the stable release of Xamarin Forms is used. I have also removed the PCL output of the library as this caused issues in Visual Studio Mac

JimSEOW commented:
@Mark-RSK 

Error	NU1202	Package Microsoft.TestPlatform.TestHost 15.0.0 is not compatible with netstandard1.4 (.NETStandard,Version=v1.4). Package Microsoft.TestPlatform.TestHost 15.0.0 supports:
  - net45 (.NETFramework,Version=v4.5)
  - netstandard1.5 (.NETStandard,Version=v1.5)

JimSEOW commented:
@Mark-RSK  Why I have to keep the FreshMVVM to .NET standard 1.0 and not 1.4 (causing errors in TinyIoC.cs)

Mark-RSK commented:
@JimSEOW It seems TinyIoC hasn't been updated in a while and the ifdefs were incorrect for net standard 1.4. I have updated the ifdef's to include 1.3 & 1.4 and it now all compiles.

I apologise for the earlier checkin that failed to build. I had changed the csproj type of the test project from netstandard to net461 on my master branch but hadn't made the change in this branch.

jfversluis commented:
I am waiting for this! Nice work :)

JimSEOW commented:
@Mark-RSK thanks.

JimSEOW commented:
@Mark-RSK   These are the failed tests
![2017-08-11_19-01-32](https://user-images.githubusercontent.com/8244633/29223660-a1ceac0e-7ec7-11e7-9e45-73b957d3db2a.png)

Mark-RSK commented:
@JimSEOW I've fixed the failing tests. Not sure the tests would have passed on the original master branch. I am unable to build the original master to check

Mark-RSK commented:
Building the original branch in VS2015 and running the tests, the same tests fail

rid00z commented:
I'll take a look at integrating this asap. I'm away from computer for the
next week but I can take a look at this on return.

Thanks


On Sat, 12 Aug 2017 at 4:59 pm, Mark Pitt <notifications@github.com> wrote:

> Building the original branch in VS2015 and running the tests, the same
> tests fail
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rid00z/FreshMvvm/pull/170#issuecomment-321967629>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABeXvQwcz8zjWlpud14X70s_7Od9qiSjks5sXWmMgaJpZM4OTQJ4>
> .
>
-- 

[image: XAM-Consulting Pty Ltd] <http://www.xam-consulting.com/>

[image: Twitter]  <https://twitter.com/XAMConsulting>   [image: LinkedIn]
<https://www.linkedin.com/company-beta/5791295/>

Michael Ridland / Director / Xamarin & Microsoft MVP
michael@xam-consulting.com / 0404 865 350

XAM Consulting Pty. Limited.
Web | Cloud | Apps | Xamarin
210/29 Kiora Rd, Miranda NSW 2228
xam-consulting.com <http://www.xam-consulting.com/>

daveclarke commented:
Without wishing to be rude/pushy, is there any progress on merging this PR? Thanks :-)

rid00z commented:
@daveclarke @Mark-RSK @jfversluis Apologies for the slow reply, we've upgrade to .NET Standard already but there's a few nice commits/ideas in here that I will take a look at.
